### ISSUE TITLE
What options do i have to track parallel executed work. Can we add some

### ISSUE THREAD

**Issue by xMAC94x** [rayon-rs/rayon#832]
I want to track the time i spend in my code. so i create a `Instant` before and afterwards and i know how much time i spend.
Now i use rayon to improve my code and do the same.
However, i don't know how many cores my code ran on.
Okay so I could assume it was running on all available cores, but what if i do another task at the same time ?
Did task1 get scheduled at 50% of all cores and task2 at the other 50% ?
Now imagine more and more tasks, all running at the same time doing different important code in parallel thanks to Rayon.

It would be nice to get some feedback with the result. E.g. Your code was scheduled on 4.3 cores on average during the runtime of the iteration.

This way we in our veloren game engine could find out not only the slowest path but how much CPU time is burned there and where to optimize.

Or is there already some feedback like this ?

More detailed feedback, like the iterator first run on 8 cores, then after 10 seconds it had to share some cores and after another 6 seconds it ran on all cores again would also be possible, but prob to much overhead?

Doing a measurement in every single iteration is prob a way to big overhead. Doing one per chunk of items would prob work

**Comment by wagnerf42** [comment]
hi,

i have a fork of rayon which enables some logs.
it is experimental but if you like you can give it try.

there is an example [here](https://github.com/wagnerf42/fast-tracer/blob/master/examples/rayon.rs)

note that the [corresponding Cargo.toml file](https://github.com/wagnerf42/fast-tracer/blob/master/Cargo.toml) is using my modified rayon and not the official one.

the svg files produced should be seen in firefox since they contain some javascript.
it shows the decomposition into tasks with each thread having a different color.
